crates/dekaf/src/log_appender.rs

```diff
@@ -51,6 +51,21 @@ impl StatsAggregator {
         if let Some(out) = &stats.out {
             ops::merge_docs_and_bytes(out, &mut binding.out);
         }
+        if let Some(last_source_published_at) = stats.last_source_published_at {
+            binding.last_source_published_at = Some(match binding.last_source_published_at {
+                Some(prev) => {
+                    if prev.seconds < last_source_published_at.seconds
```

Comment:
I don't understand why this check would be needed. Why would `last_source_published_at` ever be less than `prev`? Is it just when we backfill? My knee jerk is that this condition doesn't really seem right.

Reply:
`StatsAggregator`'s job is to roll up a bunch of stats events into a single stats event so it can be emitted periodically. My thinking here is, just like how we merge in the `DocsAndBytes` above for `left`, `right`, and `out`, we should also merge in `last_source_published_at` by always selecting the sooner of the `last_source_published_at`s, so the emitted stats document always has the soonest value.

I suppose if we make the assumption that all of the emitted stats documents are in order, and remain in order through the whole codepath then this check isn't needed, but it seemed like cheap insurance in case that's not the case

Reply:
> if we make the assumption that all of the emitted stats documents are in order, and remain in order through the whole codepath

The intent of `last_source_published_at` is that it should be the `flow_published_at` of the last document that was materialized. That's how we treat it elsewhere (we don't have any similar conditionals for regular materialzations), and I think it makes sense for dekaf to treat it the same. If we somehow manage to read or materialize documents out of order then it seems like that would be its own problem, and it probably wouldn't help to have this logic potentially paper over that issue.

Reply:
Alright, sounds good I'll update this to just overwrite with the new val every time whatever it is 👍